### DISCUSSION TITLE
[2-1]

### DISCUSSION THREAD

**Discussion by moonuckjoo** [bbang-jun/2023-OOP-Mentoring#14] [Q&A]
#define _CRT_SECURE_NO_WARNINGS
#include <iostream>
#include <cstring>
using namespace std;
#include <Windows.h>

#define _CRTDBG_MAP_ALLOC
#include <cstdlib>
#include <crtdbg.h>

#ifdef _DEBUG
#define new new ( _NORMAL_BLOCK , __FILE__ , __LINE__ )
#endif

class Student_info {
private:
	char name[10];
	int age = 0;
	char university[100];
	char major[100];

public: // 값을 입력받아서 private에 저장
	Student_info(char* name, int age, char* univ, char* major)
	{
		strcpy(this->name, name);
		strcpy(this->university, univ);
		strcpy(this->major, major);
		this->age = age;
	};
	bool find(char* name)
	{
		if (strcmp(this->name, name) == 0)
			return 0;
		else
			return 1;
	};
	void print()  //private의 저장된 것 출력
	{
		cout << "----------print----------" << endl;
		cout << "Name : " << this->name << endl;
		cout << "Age : " << this->age << endl;
		cout << "University : " << this->university << endl;
		cout << "Major : " << this->major << endl;
		cout << "-------------------------" << endl;
	};
	void change(char* name, int age, char* univ, char* major)
	{
			strcpy(this->name, name);
			strcpy(this->university, univ);
			strcpy(this->major, major);
			this->age = age;
	};
	~Student_info() {
	};
	};
	int main (void)
	{
		Student_info* Student[10] = {0};
		char input[10] ={0}; // 커맨드 저장
			// 클래스 메소드에 전달할 정보 저장
		char mainname[10] = { 0 };
		int mainage = 0;
		char mainuniv[100] = { 0 };
		char mainmajor[100] = { 0 };
		char findname[10] = { 0 };
		int i = 0;
		
		while (1) // 무한 반복
		{
			cout << "Please Enter Command(Insert, find, change, print, exit) : ";
			cin >> input; // 커맨드를 입력받음
			if (strcmp(input,"insert")==0) // insert인 경우
			{	
				cin >> mainname >> mainage >> mainuniv >> mainmajor;
				Student[i] = new Student_info(mainname, mainage, mainuniv, mainmajor); 
				i++; // 다음 사람거
			}
			else if (strcmp(input, "find") == 0)
			{
				cin >> findname;
				for (int j = 0; j < i; j++) {
					if (Student[j]->find(findname) == 0)
						Student[j]->print();
				}
			}
			else if (strcmp(input, "change")==0)
			{
				cin >> findname >> mainname >> mainage >> mainuniv >> mainmajor;
				for (int l = 0; l < i; l++)
				{
					Student[l]->find(findname);
					if (Student[l]->find(findname) == 0)
					{
						Student[l]->change(mainname, mainage, mainuniv, mainmajor);
					}
				}
			}
			else if (strcmp(input, "print") == 0)
			{
				for(int k=0;k<i;k++)
					Student[k]->print();
			}
			else if (strcmp(input, "exit")==0)
				break;

			else
			{
				cout << "Please Enter Command(insert, find, change, print, exit) : ";
				cin >> input; // 커맨드를 입력받음
			}
		}
		delete Student;
		_CrtDumpMemoryLeaks();
		return 0;
	}
![image](https://user-images.githubusercontent.com/108858978/216520121-d7efa3eb-0792-4cdf-a00d-a0cd1f0bfd7b.png)
다른 기능은 다 잘 되는데 exit를 타이핑하면 .exe가 중단점을 트리거했다면서 에러가 나네요 ㅠㅠ 갑자기 왜 이럴까요?
검색해보니까 동적할당된 메모리보다 더 크게 초기화하는 경우 발생한다는데 잘 이해가 안되네요... 이게 문제라면 다른 기능에 다 문제가 있었어야 하는게 아닐까 싶기도 하구요ㅜ

**Comment by Bjimin** [comment]
문제 상황에 대한 정보가 부족하여 제가 말씀해주시는 문제 상황을 제대로 이해했는지는 모르겠습니다.
그러나 과제가 오늘까지라고 들었기 때문에 우선 답변을 드리면, 아래에 질문해주신 서은님의 질문 내용과 유사한 문제가 존재하는 것 같습니다. 따라서 아래 질문과 답변을 읽어주신 후 메모리 해제 부분을 다시 작성해보시길 바랍니다.
문제 상황과는 별개로, 동적할당을 할 수 있는 공간을 10개까지 선언해놓으셨는데 i가 10 이상인 경우, 제한하는 로직은 어디에 두신건지 여쭤보고 싶습니다.
이후 여전히 문제가 발생한다면 추가질문 해주시길 바랍니다!
